C/common/management_client.cpp

```diff
@@ -368,6 +368,62 @@ string payload;
  * @param category	The name of the category to register
  * @return bool		True if the registration was succesful
  */
+bool ManagementClient::registerCategoryChild(const string& category)
+{
+ostringstream convert;
+
+	if (m_uuid == 0)
+	{
+		// Not registered with core
+		m_logger->error("Service is not registered with the core - not registering configuration interest");
+		return true;
+	}
+	try {
+		convert << "{ \"category\" : \"" << JSONescape(category) << "\", ";
+		convert << "\"child\" : \"" << "True" << "\", ";
+		convert << "\"service\" : \"" << *m_uuid << "\" }";
+
+		auto res = this->getHttpClient()->request("POST", "/fledge/interest", convert.str());
+		Document doc;
+		string content = res->content.string();
+		doc.Parse(content.c_str());
+		if (doc.HasParseError())
+		{
+			bool httpError = (isdigit(content[0]) && isdigit(content[1]) && isdigit(content[2]) && content[3]==':');
+			m_logger->error("%s category registration: %s\n",
+								httpError?"HTTP error during":"Failed to parse result of",
+								content.c_str());
+			return false;
+		}
+		if (doc.HasMember("id"))
+		{
+			const char *reg_id = doc["id"].GetString();
+			m_categories[category] = string(reg_id);
+			m_logger->info("Registered configuration category %s, registration id %s.",
+					category.c_str(), reg_id);
+			return true;
+		}
+		else if (doc.HasMember("message"))
+		{
+			m_logger->error("Failed to register configuration category: %s.",
```

Comment:
It might be worth adding "child" to the message so we can distinguish it from the routine below.